File: sdk/dart/lib/api/product_lookup_api.dart
part of openapi.api;



class ProductLookupApi {
  final ApiClient apiClient;

  ProductLookupApi([ApiClient apiClient]) : apiClient = apiClient ?? defaultApiClient;

  /// Find item by asin code with HTTP info returned
  ///
  /// Returns a single item
  Future<Response> getItemByASINWithHttpInfo(String asin) async {
    Object postBody;

    // verify required params are set
    if(asin == null) {
     throw ApiException(400, "Missing required param: asin");
    }

    // create path and map variables
    String path = "/api/v1/asin/{asin}".replaceAll("{format}","json").replaceAll("{" + "asin" + "}", asin.toString());

    // query params
    List<QueryParam> queryParams = [];
    Map<String, String> headerParams = {};
    Map<String, String> formParams = {};

    List<String> contentTypes = [];

    String contentType = contentTypes.isNotEmpty ? contentTypes[0] : "application/json";
    List<String> authNames = [];

    if(contentType.startsWith("multipart/form-data")) {
      bool hasFields = false;
      MultipartRequest mp = MultipartRequest(null, null);
      if(hasFields)
        postBody = mp;
    }
    else {
    }

    var response = await apiClient.invokeAPI(path,
                                             'GET',
                                             queryParams,
                                             postBody,
                                             headerParams,
                                             formParams,
                                             contentType,
                                             authNames);
    return response;
  }

  /// Find item by asin code
  ///
  /// Returns a single item
  Future<Item> getItemByASIN(String asin) async {
    Response response = await getItemByASINWithHttpInfo(asin);
    if(response.statusCode >= 400) {
      throw ApiException(response.statusCode, _decodeBodyBytes(response));
    } else if(response.body != null) {
      return apiClient.deserialize(_decodeBodyBytes(response), 'Item') as Item;
    } else {
      return null;
    }
  }

  /// Find item by UPC code with HTTP info returned
  ///
  /// Returns a single item
  Future<Response> getItemByEANWithHttpInfo(String ean) async {
    Object postBody;

    // verify required params are set
    if(ean == null) {
     throw ApiException(400, "Missing required param: ean");
    }

    // create path and map variables
    String path = "/api/v1/ean/{ean}".replaceAll("{format}","json").replaceAll("{" + "ean" + "}", ean.toString());

    // query params
    List<QueryParam> queryParams = [];
    Map<String, String> headerParams = {};
    Map<String, String> formParams = {};

    List<String> contentTypes = [];

    String contentType = contentTypes.isNotEmpty ? contentTypes[0] : "application/json";
    List<String> authNames = [];

    if(contentType.startsWith("multipart/form-data")) {
      bool hasFields = false;
      MultipartRequest mp = MultipartRequest(null, null);
      if(hasFields)
        postBody = mp;
    }
    else {
    }

    var response = await apiClient.invokeAPI(path,
                                             'GET',
                                             queryParams,
                                             postBody,
                                             headerParams,
                                             formParams,
                                             contentType,
                                             authNames);
    return response;
  }

  /// Find item by UPC code
  ///
  /// Returns a single item
  Future<Item> getItemByEAN(String ean) async {
    Response response = await getItemByEANWithHttpInfo(ean);
    if(response.statusCode >= 400) {
      throw ApiException(response.statusCode, _decodeBodyBytes(response));
    } else if(response.body != null) {
      return apiClient.deserialize(_decodeBodyBytes(response), 'Item') as Item;
    } else {
      return null;
    }
  }

  /// Find item by UPC code with HTTP info returned
  ///
  /// Returns a single item
  Future<Response> getItemByUPCWithHttpInfo(String upc) async {
    Object postBody;

    // verify required params are set
    if(upc == null) {
     throw ApiException(400, "Missing required param: upc");
    }

    // create path and map variables
    String path = "/api/v1/upc/{upc}".replaceAll("{format}","json").replaceAll("{" + "upc" + "}", upc.toString());

    // query params
    List<QueryParam> queryParams = [];
    Map<String, String> headerParams = {};
    Map<String, String> formParams = {};

    List<String> contentTypes = [];

    String contentType = contentTypes.isNotEmpty ? contentTypes[0] : "application/json";
    List<String> authNames = [];

    if(contentType.startsWith("multipart/form-data")) {
      bool hasFields = false;
      MultipartRequest mp = MultipartRequest(null, null);
      if(hasFields)
        postBody = mp;
    }
    else {
    }

    var response = await apiClient.invokeAPI(path,
                                             'GET',
                                             queryParams,
                                             postBody,
                                             headerParams,
                                             formParams,
                                             contentType,
                                             authNames);
    return response;
  }

  /// Find item by UPC code
  ///
  /// Returns a single item
  Future<Item> getItemByUPC(String upc) async {
    Response response = await getItemByUPCWithHttpInfo(upc);
    if(response.statusCode >= 400) {
      throw ApiException(response.statusCode, _decodeBodyBytes(response));
    } else if(response.body != null) {
      return apiClient.deserialize(_decodeBodyBytes(response), 'Item') as Item;
    } else {
      return null;
    }
  }

}

File: sdk/csharp-dotnet2/src/main/CsharpDotNet2/Org/OpenAPITools/Api/ProductLookupApi.cs
using System;
using System.Collections.Generic;
using RestSharp;
using Org.OpenAPITools.Client;
using Org.OpenAPITools.Model;

namespace Org.OpenAPITools.Api
{
    /// <summary>
    /// Represents a collection of functions to interact with the API endpoints
    /// </summary>
    public interface IProductLookupApi
    {
        /// <summary>
        /// Find item by asin code Returns a single item
        /// </summary>
        /// <param name="asin">ASIN code of item to return</param>
        /// <returns>Item</returns>
        Item GetItemByASIN (string asin);
        /// <summary>
        /// Find item by UPC code Returns a single item
        /// </summary>
        /// <param name="ean">EAN code of item to return</param>
        /// <returns>Item</returns>
        Item GetItemByEAN (string ean);
        /// <summary>
        /// Find item by UPC code Returns a single item
        /// </summary>
        /// <param name="upc">UPC code of item to return</param>
        /// <returns>Item</returns>
        Item GetItemByUPC (string upc);
    }
  
    /// <summary>
    /// Represents a collection of functions to interact with the API endpoints
    /// </summary>
    public class ProductLookupApi : IProductLookupApi
    {
        /// <summary>
        /// Initializes a new instance of the <see cref="ProductLookupApi"/> class.
        /// </summary>
        /// <param name="apiClient"> an instance of ApiClient (optional)</param>
        /// <returns></returns>
        public ProductLookupApi(ApiClient apiClient = null)
        {
            if (apiClient == null) // use the default one in Configuration
                this.ApiClient = Configuration.DefaultApiClient; 
            else
                this.ApiClient = apiClient;
        }
    
        /// <summary>
        /// Initializes a new instance of the <see cref="ProductLookupApi"/> class.
        /// </summary>
        /// <returns></returns>
        public ProductLookupApi(String basePath)
        {
            this.ApiClient = new ApiClient(basePath);
        }
    
        /// <summary>
        /// Sets the base path of the API client.
        /// </summary>
        /// <param name="basePath">The base path</param>
        /// <value>The base path</value>
        public void SetBasePath(String basePath)
        {
            this.ApiClient.BasePath = basePath;
        }
    
        /// <summary>
        /// Gets the base path of the API client.
        /// </summary>
        /// <param name="basePath">The base path</param>
        /// <value>The base path</value>
        public String GetBasePath(String basePath)
        {
            return this.ApiClient.BasePath;
        }
    
        /// <summary>
        /// Gets or sets the API client.
        /// </summary>
        /// <value>An instance of the ApiClient</value>
        public ApiClient ApiClient {get; set;}
    
        /// <summary>
        /// Find item by asin code Returns a single item
        /// </summary>
        /// <param name="asin">ASIN code of item to return</param> 
        /// <returns>Item</returns>            
        public Item GetItemByASIN (string asin)
        {
            
            // verify the required parameter 'asin' is set
            if (asin == null) throw new ApiException(400, "Missing required parameter 'asin' when calling GetItemByASIN");
            
    
            var path = "/api/v1/asin/{asin}";
            path = path.Replace("{format}", "json");
            path = path.Replace("{" + "asin" + "}", ApiClient.ParameterToString(asin));
    
            var queryParams = new Dictionary<String, String>();
            var headerParams = new Dictionary<String, String>();
            var formParams = new Dictionary<String, String>();
            var fileParams = new Dictionary<String, FileParameter>();
            String postBody = null;
    
                                                    
            // authentication setting, if any
            String[] authSettings = new String[] {  };
    
            // make the HTTP request
            IRestResponse response = (IRestResponse) ApiClient.CallApi(path, Method.GET, queryParams, postBody, headerParams, formParams, fileParams, authSettings);
    
            if (((int)response.StatusCode) >= 400)
                throw new ApiException ((int)response.StatusCode, "Error calling GetItemByASIN: " + response.Content, response.Content);
            else if (((int)response.StatusCode) == 0)
                throw new ApiException ((int)response.StatusCode, "Error calling GetItemByASIN: " + response.ErrorMessage, response.ErrorMessage);
    
            return (Item) ApiClient.Deserialize(response.Content, typeof(Item), response.Headers);
        }
    
        /// <summary>
        /// Find item by UPC code Returns a single item
        /// </summary>
        /// <param name="ean">EAN code of item to return</param> 
        /// <returns>Item</returns>            
        public Item GetItemByEAN (string ean)
        {
            
            // verify the required parameter 'ean' is set
            if (ean == null) throw new ApiException(400, "Missing required parameter 'ean' when calling GetItemByEAN");
            
    
            var path = "/api/v1/ean/{ean}";
            path = path.Replace("{format}", "json");
            path = path.Replace("{" + "ean" + "}", ApiClient.ParameterToString(ean));
    
            var queryParams = new Dictionary<String, String>();
            var headerParams = new Dictionary<String, String>();
            var formParams = new Dictionary<String, String>();
            var fileParams = new Dictionary<String, FileParameter>();
            String postBody = null;
    
                                                    
            // authentication setting, if any
            String[] authSettings = new String[] {  };
    
            // make the HTTP request
            IRestResponse response = (IRestResponse) ApiClient.CallApi(path, Method.GET, queryParams, postBody, headerParams, formParams, fileParams, authSettings);
    
            if (((int)response.StatusCode) >= 400)
                throw new ApiException ((int)response.StatusCode, "Error calling GetItemByEAN: " + response.Content, response.Content);
            else if (((int)response.StatusCode) == 0)
                throw new ApiException ((int)response.StatusCode, "Error calling GetItemByEAN: " + response.ErrorMessage, response.ErrorMessage);
    
            return (Item) ApiClient.Deserialize(response.Content, typeof(Item), response.Headers);
        }
    
        /// <summary>
        /// Find item by UPC code Returns a single item
        /// </summary>
        /// <param name="upc">UPC code of item to return</param> 
        /// <returns>Item</returns>            
        public Item GetItemByUPC (string upc)
        {
            
            // verify the required parameter 'upc' is set
            if (upc == null) throw new ApiException(400, "Missing required parameter 'upc' when calling GetItemByUPC");
            
    
            var path = "/api/v1/upc/{upc}";
            path = path.Replace("{format}", "json");
            path = path.Replace("{" + "upc" + "}", ApiClient.ParameterToString(upc));
    
            var queryParams = new Dictionary<String, String>();
            var headerParams = new Dictionary<String, String>();
            var formParams = new Dictionary<String, String>();
            var fileParams = new Dictionary<String, FileParameter>();
            String postBody = null;
    
                                                    
            // authentication setting, if any
            String[] authSettings = new String[] {  };
    
            // make the HTTP request
            IRestResponse response = (IRestResponse) ApiClient.CallApi(path, Method.GET, queryParams, postBody, headerParams, formParams, fileParams, authSettings);
    
            if (((int)response.StatusCode) >= 400)
                throw new ApiException ((int)response.StatusCode, "Error calling GetItemByUPC: " + response.Content, response.Content);
            else if (((int)response.StatusCode) == 0)
                throw new ApiException ((int)response.StatusCode, "Error calling GetItemByUPC: " + response.ErrorMessage, response.ErrorMessage);
    
            return (Item) ApiClient.Deserialize(response.Content, typeof(Item), response.Headers);
        }
    
    }
}
